### ISSUE TITLE
Thoughts on support TinyStr16

### ISSUE THREAD

**Issue by Swoorup** [drifting-in-space/block-id#1]
It would be great if we could directly allocated on the stack using TinyStr16 or fixed string.

**Comment by paulgb** [comment]
Is the goal here to remove heap allocations entirely? If so there's also an [intermediate Vec](https://github.com/drifting-in-space/block-id/blob/1b2568ddd62bb76ed34f53537aa2d9f198c12572/src/base.rs#L26) that we would want to move to the stack as well.

I'm open to this. I'm curious if you have a use case in mind that benefits measurably from this (helps to have some context for design decisions.)

**Comment by Swoorup** [comment]
> Is the goal here to remove heap allocations entirely?

That is correct. I am finding sometimes having to do a lot of upfront generation of short ids, where making it entirely allocate on stack would certainly improve some performance.

I did come across the intermediate Vec earlier, perhaps we could use `SmallVec` or `TinyVec` for that, behind a feature flag?
